### PR TITLE
Added a single test to the testing binary

### PR DESCRIPTION
**Pull request by fhaynes**
Also added more options to the CLI

### CODE REVIEW

File: src/bin/laminar-tester.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+use std::process::exit;
+use std::thread;
+use std::time::{Instant, Duration};
+use std::net::SocketAddr;
+
+use clap::App;
+
+use laminar::net;
+use laminar::packet::Packet;
+use laminar::infrastructure::delivery_method::DeliveryMethod;
```
[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/laminar-tester.rs]: Ahh, you are right. I missed that. Also removes the need for making the mod pub.

File: src/bin/cli.yml [hunk 2]
```diff
@@ -2,50 +2,124 @@ name: tester
...
+                short: h
+            - LISTEN_PORT:
+                help: "What port the server should listen on"
+                required: true
+                takes_value: true
+                long: bind-port
+                default_value: "2264"
+                short: p
+            - SHUTDOWN_TIMER:
+                help: The number of seconds the server will remain running, then shut itself down
```
[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/cli.yml]: YAML doesn't usually require quotes around strings, but I do it out of habit.

**Comment by fhaynes** [comment]
I agree with the idea of a config struct, was going to do that next round.

File: src/bin/cli.yml [hunk 3]
```diff
@@ -2,50 +2,124 @@ name: tester
...
-      long: connect-host
-      short: h
-    CONNECT_PORT:
-      help: "What port the client will connect to"
-      required: true
-      takes_value: true
-      long: connect-port
-      short: p
-    TEST_TO_RUN:
-      help: "Which client test to run. Run laminar-tester show-tests to see all available"
```
[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/cli.yml]: Changed, but it didn't update the PR for some reason.

File: src/bin/cli.yml [hunk 4]
```diff
@@ -2,50 +2,124 @@ name: tester
...
-      long: bind-host
-      short: h
-    - LISTEN_PORT:
-      help: "What port the server should listen on"
-      required: true
-      takes_value: true
-      long: bind-port
-      short: p
-    - SHUTDOWN_TIMER:
-      help: The number of seconds the server will remain running, then shut itself down
```
[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/cli.yml]: Fixed, but it didn't update the PR.

File: src/bin/cli.yml [hunk 5]
```diff
@@ -2,50 +2,124 @@ name: tester
...
+                default_value: "0"
+                short: p
+            - TEST_TO_RUN:
+                help: "Which client test to run. Run `laminar-tester show-tests` to see all available"
+                required: true
+                takes_value: true
+                long: test
+                short: t
+                default_value: "steady-stream"
+            - CLIENT_TIMEOUT:
```
[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/cli.yml]: I removed the CLIENT_TIMEOUT as I wasn't using it.

File: src/bin/laminar-tester.rs [hunk 6]
```diff
@@ -0,0 +1,123 @@
+#[macro_use]
```
[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/laminar-tester.rs]: Not sure what you mean...just that the binary shouldn't be compiled automatically?

[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/laminar-tester.rs]: And why not just build with `cargo build --lib`?

[the remark this answers is not in the document]
Comment (LucioFranco) [reply src/bin/laminar-tester.rs]: This should be solved via the feature gate.

[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/laminar-tester.rs]: That doesn't answer my question of what that solves that building with `cargo build --lib` doesn't.

[the remark this answers is not in the document]
Comment (fhaynes) [reply src/bin/laminar-tester.rs]: This is resolved by the other feature gate additions.

**Pull request merged**